### DISCUSSION TITLE
Image Resolution limit for responsive layout

### DISCUSSION THREAD

**Discussion by rinkusam12** [vercel/next.js#25231] [Help]
Does anyone know how to limit the max image resolution that can be loaded on a "responsive" layout for the next/image component?
Currently, even for thumbnails, it loads a  very heavy resolution image that isn't very optimized and causes a huge load.

**Comment by bobarros** [comment]
Change the default max value (below) to be higher or smaller than 3840 in the next js config file.

```javascript
module.exports = {
  images: {
    deviceSizes: [640, 750, 828, 1080, 1200, 1920, 2048, 3840],
  },
}
```

For thumbnails, it's recommended to use layout="fixed" or layout="intrinsic" though. 
Responsive is for larger images.
To limit them, you would have to use different options. Below, the default.

```javascript
module.exports = {
  images: {
    imageSizes: [16, 32, 48, 64, 96, 128, 256, 384],
  },
}
```

**Reply by rinkusam12** [reply]
> Change the default max value (below) to be higher or smaller than 3840 in the next js config file.
> 
> ```js
> module.exports = {
>   images: {
>     deviceSizes: [640, 750, 828, 1080, 1200, 1920, 2048, 3840],
>   },
> }
> ```
> 
> For thumbnails, it's recommended to use layout="fixed" or layout="intrinsic" though.
> Responsive is for larger images.
> To limit them, you would have to use different options. Below, the default.
> 
> ```js
> module.exports = {
>   images: {
>     imageSizes: [16, 32, 48, 64, 96, 128, 256, 384],
>   },
> }
> ```

This setting affects everything on a global level which isn't a very optimal solution. Using layout="intrinsic" or layout="fixed", limit image expansion, So, the solution that I am looking for is using the responsive setting I somehow want to limit the size to some specific resolution so it don't load heavy size resolution.

Problem using layout intrinsic:
https://i.ibb.co/HKRrbBq/layout-intrinsic.png, As you I need the image to fill up but it caused white space,

In responsive it work perfectly but the issue is it loads up heavy image size.

**Reply by macrozone** [reply]
i am also very bothered by this. I wanted to migrate to the new Image component. I used `layout="fill"` first, but that loads huge images, then I tried `layout="responsive", but that does not seem to make any difference, even if i specify a size. 

`layout="intrinsic"` works best, but that leaves whitespace on bigger screens (i would rather have it to be a bit blurry, because we only hit that on edge cases when switching between 1 and two columns responsivly).

i would expect layout="responsive" to honor the size you give, not just loading the biggest size possible.
